packages/legacy/core/App/hooks/connections.ts

```diff
@@ -9,3 +9,21 @@ export const useConnectionByOutOfBandId = (outOfBandId: string): ConnectionRecor
     [connections, outOfBandId]
   )
 }
+
+export const useOutOfBandById = (oobId: string): OutOfBandRecord | undefined => {
+  const { agent } = useAgent()
+  const [oob, setOob] = useState<OutOfBandRecord | undefined>(undefined)
+  if (!oob) {
+    agent?.oob.findById(oobId).then((res) => {
+      if (res) {
+        setOob(res)
+      }
+    })
+  }
+  return useMemo(() => oob, [oobId, oob])
```

Comment:
not really sure the `useMemo `is really made use of here, and because the logic isn't in a `useEffect` I don't think it will respond to the `oobId` changing and update `oob`. (Not sure if we need that functionality anyway but still)

Reply:
initially oob is undefined, I used UseMemo so that once they are defined then the function updates